src/functions-and-arrays.js

```diff
@@ -1,38 +1,68 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  } else {
+    return number2;
+  }
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(wordarray) {
+  if (wordarray.length === 0) {
+    return null;
+  }
+
+  let longestword = '';
+  for (let i = 0; i < wordarray.length; i++) {
+    if (wordarray[i].length > longestword.length) {
+      longestword = wordarray[i];
+    }
+  }
+  return longestword;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
-
+function sumNumbers(numbersarray) {
+  let result = 0;
+  for (let i = 0; i < numbersarray.length; i++) {
+    result = result + numbersarray[i];
+  }
+  return result;
+}
 
 // Iteration #3.1 Bonus:
 function sum() {}
 
-
-
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(numbersarray) {
+  if (numbersarray.length === 0) {
+    return null;
+  }
+let sum = sumNumbers (numbersarray)
+let numberOfElements= numbersarray.length
+let average = sum / numberOfElements
+return average
```

Comment:
Another tiny detail. Just so that you know, you could do this in one line `return sumNumbers(numbersarray) / numbersarray.length`